app/charts/map/map.tsx

```diff
@@ -160,8 +161,19 @@ export const MapComponent = () => {
       return;
     }
 
-    if (bbox && !isViewStateLocked) {
-      setViewState(constrainZoom(viewState, bbox));
+    let newViewState: MinMaxZoomViewState | undefined = undefined;
+
+    if (bbox) {
+      if (!isViewStateLocked) {
+        newViewState = constrainZoom(viewState, bbox);
+      } else {
+        newViewState = viewState;
+      }
+    }
+
+    if (newViewState) {
+      initialViewStateRef.current = newViewState;
```

Comment:
Since we update the ref here, it does not seem that "initial" is the right word here (I would expect that a variable named with initial would not change during the lifecycle of the component).

I find this useEffect a bit hard to understand, do you think it would be easier to understand if it was extracted with a good name ? What is the end goal of this useEffect ?

Reply:
I'll refactor the code, thanks for suggestions 🙇